src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -23,10 +25,104 @@ private static string GetGitSystemHash(GitPreparer gitPreparer, IFileSystem file
         {
             var dotGitDirectory = gitPreparer.GetDotGitDirectory();
 
-            // Maybe using timestamp in .git/refs directory is enough?
-            var lastGitRefsChangedTicks = fileSystem.GetLastDirectoryWrite(Path.Combine(dotGitDirectory, "refs"));
+            // traverse the directory and get a list of files, use that for GetHash
+            var traverse = TraverseTree(Path.Combine(dotGitDirectory, "refs"));
 
-            return GetHash(dotGitDirectory, lastGitRefsChangedTicks.ToString());
+            return GetHash(traverse.ToArray());
+        }
+
+        // lifted from https://msdn.microsoft.com/en-us/library/bb513869.aspx
+        public static List<string> TraverseTree(string root)
+        {
+            var result = new List<string> { root };
+
+            // Data structure to hold names of subfolders to be
+            // examined for files.
+            var dirs = new Stack<string>(20);
```

Comment:
http://stackoverflow.com/questions/1762817/default-capacity-of-list talks about List though I suspect it will be applicable. Starts at capacity 0, then 4, then doubles as space is needed. 20 seems reasonable to me, but we can start at 0 if you want.